src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -384,4 +430,323 @@ private static String trimExcessWhitespaces(String text) {
         return result.toString();
     }
 
+    /**
+     * Finds and return if summary tag present.
+     *
+     * @param javadoc Javadoc root node.
+     * @return true, if first sentence contains @summary tag.
+     */
+    private static boolean containsSummaryTag(DetailNode javadoc) {
+        boolean contains = false;
+        for (DetailNode node : javadoc.getChildren()) {
+            if (node.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                final DetailNode[] child = node.getChildren();
+                if (child[1].getType() == JavadocTokenTypes.CUSTOM_NAME
+                    && SUMMARY_TEXT.equals(child[1].getText())) {
+                    contains = true;
+                }
+            }
+        }
+
+        if (containsSummaryTagWithHtmlTag(javadoc)) {
+            contains = true;
+        }
+
+        return contains;
+    }
+
+    /**
+     * Finds and return if summary tag present as HTML Format.
+     *
+     * @param javadoc Javadoc root node.
+     * @return true, if first sentence contains @summary tag.
+     */
+    private static boolean containsSummaryTagWithHtmlTag(DetailNode javadoc) {
+        boolean contains = false;
+        if (getInlineTagNodeInsideHtml(javadoc) != null) {
+            final DetailNode[] htmlChild = getInlineTagNodeInsideHtml(javadoc).getChildren();
+            contains = htmlChild[1].getType() == JavadocTokenTypes.CUSTOM_NAME
```

Comment:
it is not safe to do `[1]` without checking for size

Reply:
Same, as above, it is not required, since htmlChild contains children of Inline Tag, at atleast 2 children will be present.

Reply:
please move this explanation as comment over this code.
ti is not very obvious.

Reply:
Done